tests/ui/async-await/for-await.rs

```diff
@@ -0,0 +1,24 @@
+// run-pass
+// edition: 2021
+#![feature(async_iterator, async_iter_from_iter, const_waker, async_for_loop, noop_waker)]
+
+use std::future::Future;
+
+// make sure a simple for await loop works
+async fn real_main() {
```

Comment:
Please add a test that this works correctly in an async-gen fn as well. Maybe just one that does:

```
for await i in iter { yield i; }
```

Or something to pass-through the awaited values.

Reply:
That turned out to be really helpful because I realized my desugaring wasn't pinning the iterator so to use it with `async gen fn` you'd have to use the `pin!` yourself, which seems pretty unfortunate.